src/app/protocol.ts

```diff
@@ -223,14 +176,14 @@ class XpraProtocol {
     while (this.websocket && this.do_process_receive_queue());
   }
 
-  error() {
+  error(...args) {
     console.error.apply(console, arguments);
   }
-  log() {
+  log(...args) {
     console.log.apply(console, arguments);
   }
 
-  do_process_receive_queue() {
+  do_process_receive_queue(): any {
```

Comment:
Why not void?

Reply:
This function returns `false` in some cases, so typing this as void would lead to an error.